menu.lisp

```diff
@@ -54,37 +54,49 @@
           m)))
 
 (defstruct menu-state
-  table prompt selected view-start view-end
+  unfiltered-table table prompt selected view-start view-end
   (current-input (make-array 10 :element-type 'character :adjustable t :fill-pointer 0)))
 
+(defun menu-scrolling-required (menu)
+  (and *menu-maximum-height*
+       (> (length (menu-state-table menu))
+          *menu-maximum-height*)))
+
+(defun menu-height (menu)
+  (let ((len (length (menu-state-table menu))))
+    (min (or *menu-maximum-height* len) len)))
+
 (defun bound-check-menu (menu)
   "Adjust the menu view and selected item based
 on current view and new selection."
   (let ((len (length (menu-state-table menu))))
+    ;; Wrap around
     (setf (menu-state-selected menu)
           (cond ((< (menu-state-selected menu) 0) (1- len))
                 ((>= (menu-state-selected menu) len) 0)
                 (t (menu-state-selected menu))))
-    (when (and *menu-maximum-height*
-               (> len *menu-maximum-height*))  ; scrolling required
-      (let ((sel (menu-state-selected menu)))
+    (if 0
```

Comment:
This doesn't make any sense. `(IF 0 <THEN> <ELSE>)` will always evaluate `<THEN>`, so  this entire `IF` is really just the `<THEN>`:

```lisp
(setf (menu-state-view-start menu) 0
      (menu-state-view-end menu) (length (menu-state-table menu)))
```